### PR TITLE
Update fingerprint when tree depth is changed

### PR DESCRIPTION
**Pull request by vplasencia**
## Related Issue

#197

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: apps/api/src/app/groups/groups.service.ts [hunk 1]
```diff
@@ -142,6 +142,16 @@ export class GroupsService {
 
         await this.groupRepository.save(group)
 
+        if (treeDepth) {
```
Comment (cedoor) [inline apps/api/src/app/groups/groups.service.ts:145]: I think we can merge this if block and the other one at line 126. What do you think?

Comment (vplasencia) [reply apps/api/src/app/groups/groups.service.ts:145]: Yes, true. Thanks! 👍

File: apps/api/src/app/groups/groups.service.test.ts [hunk 2]
```diff
@@ -98,6 +98,19 @@ describe("GroupsService", () => {
...
+            const newTreeDepth = 20
+            const { treeDepth } = await groupsService.updateGroup(
+                groupId,
+                {
+                    treeDepth: newTreeDepth
+                },
+                "admin"
+            )
+
+            expect(treeDepth).toBe(newTreeDepth)
```
Comment (saleel) [inline apps/api/src/app/groups/groups.service.test.ts:112]: A better test would be to make sure the group was actually updated in the contract, like in [here](https://github.com/privacy-scaling-explorations/bandada/blob/876c69b03c0dd694f15116abbc3c01671ecdbe78/apps/api/src/app/groups/groups.service.test.ts#LL195C11-L198C33)

Right now we are only checkin the treeDepth value is set, which would have passed even without the new change introduced in this PR.

Comment (vplasencia) [reply apps/api/src/app/groups/groups.service.test.ts:112]: Yes. I will add those lines too. Thanks! 👌

**Review by cedoor — APPROVED** [review]

**Pull request merged**